SPECS/kubernetes/CVE-2025-31133.patch

```diff
@@ -0,0 +1,439 @@
+From 8476df83b534a2522b878c0507b3491def48db9f Mon Sep 17 00:00:00 2001
+From: Kir Kolyshkin <kolyshkin@gmail.com>
+Date: Thu, 6 Mar 2025 08:19:45 -0800
+Subject: [PATCH] libct: add/use isDevNull, verifyDevNull
+
+The /dev/null in a container should not be trusted, because when /dev
+is a bind mount, /dev/null is not created by runc itself.
+
+1. Add isDevNull which checks the fd minor/major and device type,
+   and verifyDevNull which does the stat and the check.
+
+2. Rewrite maskPath to open and check /dev/null, and use its fd to
+   perform mounts. Move the loop over the MaskPaths into the function,
+   and rename it to maskPaths.
+
+3. reOpenDevNull: use verifyDevNull and isDevNull.
+
+4. fixStdioPermissions: use isDevNull instead of stat.
+
+Fixes: GHSA-9493-h29p-rfm2 CVE-2025-31133
+Co-authored-by: Rodrigo Campos <rodrigoca@microsoft.com>
+Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>
+Signed-off-by: Aleksa Sarai <cyphar@cyphar.com>
+
+Upstream Patch Reference: https://github.com/opencontainers/runc/commit/8476df83b534a2522b878c0507b3491def48db9f.patch
```

Comment:
IMHO - we should consider:
- https://github.com/opencontainers/runc/commit/2f74f4ae1b483b3a4d197f574a37c7b768baf96f

instead of the specific patch that is mentioned.
I am taking a deeper look though.